tools/test-api

```diff
@@ -42,14 +44,24 @@ with test_server_running(force=options.force, external_host='zulipdev.com:9981')
     user = get_user(email, realm)
     api_key = get_api_key(user)
     site = 'http://zulip.zulipdev.com:9981'
-
     client = Client(
         email=email,
         api_key=api_key,
         site=site
     )
 
-    # Prepare the admin client
+    # Prepare the bot client
+    email = 'outgoing-webhook@zulip.com'
+    realm = get_realm("zulip")
+    bot_user = get_user(email, realm)
+    api_key = get_api_key(user)
```

Comment:
Your bug is that you should be passing `bot_user` here -- you were sending Iago's API key.  

I figured this out by just print-debugging the code path giving the error.

Reply:
:man_facepalming: Oh wow. This is a ridiculously stupid mistake on my part, I'm surprised that I didn't notice this earlier. Sorry for wasting your time here @timabbott.

Reply:
This sort of issue is why I'm always careful to double-check my work after doing a copy-paste-and-edit; one can spend a lot of time debugging code that one would never have written intentionally :)